### PR TITLE
docs: Generate configuration guide in mvn build

### PR DESCRIPTION
**Pull request by andygrove**
## Which issue does this PR close?

Closes https://github.com/apache/datafusion-comet/issues/315

## Rationale for this change

We should publish our configuration settings so that users know about them.

## What changes are included in this PR?
Screenshot 2024-04-29 at 10 23 00 AM

## How are these changes tested?

Tested manually by running `mvn clean package -DskipTests` and then generating the documentation.

### CODE REVIEW

File: common/src/main/scala/org/apache/comet/CometConf.scala [hunk 1]
```diff
@@ -612,3 +625,33 @@ private[comet] case class ConfigBuilder(key: String) {
 private object ConfigEntry {
   val UNDEFINED = "<undefined>"
 }
+
+/**
+ * Utility for generating markdown documentation from the configs.
+ */
```
Comment (viirya) [inline common/src/main/scala/org/apache/comet/CometConf.scala:631]: Would you like to put the usage (i.e., `mvn clean package -DskipTests`) in the code comment here?

Comment (andygrove) [reply common/src/main/scala/org/apache/comet/CometConf.scala:633]: Thanks. Added.

**Review by viirya — APPROVED** [review]

File: docs/source/user-guide/configs.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+| spark.comet.columnar.shuffle.enabled | Force Comet to only use columnar shuffle for CometScan and Spark regular operators. If this is enabled, Comet native shuffle will not be enabled but only Arrow shuffle. By default, this config is false. | false |
+| spark.comet.columnar.shuffle.memory.factor | Fraction of Comet memory to be allocated per executor process for Comet shuffle. Comet memory size is specified by `spark.comet.memoryOverhead` or calculated by `spark.comet.memory.overhead.factor` * `spark.executor.memory`. By default, this config is 1.0. | 1.0 |
+| spark.comet.columnar.shuffle.spill.threshold | Number of rows to be spilled used for Comet columnar shuffle. For every configured number of rows, a new spill file will be created. Higher value means more memory requirement to buffer shuffle data before flushing to disk. As Comet uses columnar shuffle which is columnar format, higher value usually helps to improve shuffle data compression ratio. This is internal config for testing purpose or advanced tuning. By default, this config is Int.Max. | 2147483647 |
```
Comment (kazuyukitanimura) [inline docs/source/user-guide/configs.md:35]: Wondering if it is ok to publish internal conf like this one `spark.comet.columnar.shuffle.spill.threshold`...?

**Review by kazuyukitanimura — APPROVED** [review]

File: docs/source/user-guide/configs.md (hunk 2 above)
Comment (andygrove) [reply docs/source/user-guide/configs.md:35]: Great point. Let me fix that.

Comment (andygrove) [reply docs/source/user-guide/configs.md]: I pushed an update to fix this.

**Review by kazuyukitanimura — APPROVED** [review]
Thank you LGTM

**Review by viirya — APPROVED** [review]

**Comment by viirya** [comment]
Merged. Thanks @andygrove @kazuyukitanimura

**Pull request merged**